### PR TITLE
Btree dynamic entries

### PR DESCRIPTION
**Pull request by boki1**
# Description
Support dynamic entries in the B-tree.

# Test suite
```
make clean-exec-all
```

# Environment
Using gcc (GCC) 11.1.0 on a x86_64 machine with GNU/Linux.

--------------------

#### This PR makes changes which ...

- [x] are thoroughly tested
- [ ] are well documented

### CODE REVIEW

File: conanfile.txt [hunk 1]
```diff
@@ -1,13 +1,14 @@
 [requires]
-catch2/2.13.7
 fmt/8.0.1
-spdlog/1.9.2
 ms-gsl/3.1.0
+catch2/2.13.7
+spdlog/1.9.2
 libnop/cci.20200728
-cppbenchmark/cci.20201029
+andreasbuhr-cppcoro/cci.20210113
```
Comment (StoyanTinchev) [inline conanfile.txt:7]: This is 2 times in the Conan file. Why?

File: src/core/Util.h [hunk 2]
```diff
@@ -88,7 +95,7 @@ template<typename T>
 };
```
Comment (StoyanTinchev) [inline src/core/Util.h]: Remove the ";" :)

File: src/core/storage/IndirectionVector.cpp [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#include <catch2/catch.hpp>
```
Comment (StoyanTinchev) [inline src/core/storage/IndirectionVector.cpp]: Where this file is used, because it don't figure in the CmakeList

Comment (StoyanTinchev) [reply src/core/storage/IndirectionVector.cpp]: I saw where it is used, but please consider to add it in the CMakeList

File: src/core/storage/Pager.cpp [hunk 4]
```diff
@@ -1,6 +1,8 @@
 #include <array>
 #include <fstream>
 #include <vector>
+#include <algorithm>
+#include <filesystem>
```
Comment (StoyanTinchev) [inline src/core/storage/Pager.cpp]: Algorithm and filesystem imports are unused

File: src/core/storage/btree/Btree.cpp [hunk 5]
```diff
@@ -265,7 +273,7 @@ TEST_CASE("Btree persistence", "[btree]") {
 	fs::create_directories("/tmp/eugene-tests/btree-persistence");
 
 	SECTION("Loading and recovery") {
-		std::map<typename DefaultConfig::Key, typename DefaultConfig::Val> backup;
+		std::map<typename Config::Key, typename Config::Val> backup;
 		Position rootpos;
 		{
 			if constexpr (NDEBUG)
```
Comment (StoyanTinchev) [inline src/core/storage/btree/Btree.cpp:279]: I think all these checks are always correct. What is this NDEBUG?

File: src/core/storage/compression/benchmarks/CompBench.cpp [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+		st.PauseTiming();
+		clean({file_name.begin(), compressed_file_name.begin()});
+		st.ResumeTiming();
+	}
+}
+
+//	67108864 for 1GB ".txt" file -> for ~105 seconds
+//  65536 for 1MB ".txt" file -> for ~110ms
+//  64 for 1KB ".txt" file
+BENCHMARK(BM_Compression)
```
Comment (StoyanTinchev) [inline src/core/storage/compression/benchmarks/CompBench.cpp]: Why there are files related to compressor and decompressor

**Review by StoyanTinchev — CHANGES_REQUESTED** [review]
There are files related to the compressor and decompressor. Please remove this commits. Also consider the other comments.

File: src/core/Util.h (hunk 2 above)
Comment (boki1) [reply src/core/Util.h]: :+1:

File: src/core/storage/Pager.cpp (hunk 4 above)
Comment (boki1) [reply src/core/storage/Pager.cpp]: :+1:

File: src/core/storage/btree/Btree.cpp (hunk 5 above)
Comment (boki1) [reply src/core/storage/btree/Btree.cpp:279]: I want to be able to toggle debug prints. This is through the NDEBUG flag.

File: src/core/storage/IndirectionVector.cpp (hunk 3 above)
Comment (boki1) [reply src/core/storage/IndirectionVector.cpp]: It is and will remain unused. Removing it. Thanks.

**Review by StoyanTinchev — APPROVED** [review]
Good work, but please be careful now with the conflicts that need to be resolved

**Pull request merged**